internal/statemachine/common_states.go

```diff
@@ -256,6 +265,69 @@ func (stateMachine *StateMachine) populatePreparePartitions() error {
 
 // Make the disk
 func (stateMachine *StateMachine) makeDisk() error {
+	// ensure the output dir exists
+	if stateMachine.commonFlags.OutputDir == "" {
+		if stateMachine.cleanWorkDir { // no workdir specified, so create the image in the pwd
+			stateMachine.commonFlags.OutputDir, _ = os.Getwd()
+		} else {
+			stateMachine.commonFlags.OutputDir = stateMachine.stateMachineFlags.WorkDir
+		}
+	} else {
+		err := osMkdirAll(stateMachine.commonFlags.OutputDir, 0755)
+		if err != nil && !os.IsExist(err) {
+			return fmt.Errorf("Error creating OutputDir: %s", err.Error())
+		}
+	}
+	for volumeName, volume := range stateMachine.GadgetInfo.Volumes {
+		imgName := filepath.Join(stateMachine.commonFlags.OutputDir, volumeName+".img")
+
+		// Create the disk image
+		imgSize, _ := stateMachine.calculateImageSize()
+
+		diskImg, err := diskfsCreate(imgName, imgSize, diskfs.Raw)
+		if err != nil {
+			return fmt.Errorf("Error creating disk image: %s", err.Error())
+		}
+
+		// snapd always populates Schema, so it cannot be empty. Use the blocksize of the created disk
+		sectorSize := uint64(diskImg.LogicalBlocksize)
+
+		// set up the partitions on the device
+		partitionTable := createPartitionTable(volumeName, volume, sectorSize, stateMachine.isSeeded)
+
+		// Write the partition table to disk
+		if err := diskImg.Partition(*partitionTable); err != nil {
+			return fmt.Errorf("Error partitioning image file: %s", err.Error())
+		}
+
+		// TODO: go-diskfs doesn't set the disk ID when using an MBR partition table.
+		// this function is a temporary workaround, but we should change upstream go-diskfs
+		if volume.Schema == "mbr" {
+			randomBytes := make([]byte, 4)
```

Comment:
A thing to keep in mind for the future, a very unlikely edge case: since these are random bytes, does rand guarantee uniqueness of the random numbers pulled in consecutively? I guess not. So we'd probably have to safe-guard this at some point to check not to assign, in some unlucky rand roll, two same disk IDs to different volumes! Maybe keep a set of disk IDs and checking those.
I know this is temporary, so we can do that with a follow up PR.